src/components/DisplayLists.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useNavigate, useLocation } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import { FaSearch } from 'react-icons/fa';
+import AOS from 'aos';
+import 'aos/dist/aos.css';
+import fetchData, { fetchResult } from '../redux/data/fetchData';
+import './covidLists.css';
+import 'bootstrap/dist/css/bootstrap.css';
+
+const DisplayLists = () => {
+  AOS.init();
+  const results = useSelector((state) => state.result);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (results.length === 0) {
+      fetchData().then((response) => dispatch(fetchResult(response)));
+    }
+  }, [results.length, dispatch]);
+
+  let covidCountry = results.filter((result) => result.continent === 'Africa');
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
+  const search = query.get('search') || '';
+  covidCountry = covidCountry.filter((country) => country.country.includes(search));
+  const navigate = useNavigate();
+  const [searchValue, setsearchValue] = useState(search);
+
+  const filterFunction = (e) => {
+    navigate(e.target.value ? `?search=${e.target.value}` : '');
+    setsearchValue(e.target.value);
+  };
+
+  return (
+    <>
+
+      <div className="container-fluid px-4 covid-contain">
+        <ul className="row px-0">
+          <div className="col-12">
+            <h3 className="text-center">Africa</h3>
+            <form className="form d-flex justify-content-center align-items-center">
+              <div className="mb-3 border px-3 py-2">
+                <input
+                  className="search"
+                  type="text"
+                  value={searchValue}
+                  placeholder="Search country"
+                  onChange={filterFunction}
```

Comment:
- [ ] Great job implementing this filtering function :+1:  Based on the technical requirement of this project, filtering was supposed to be done in a `Filter` stateless component to filter data retrieve from the API. I'll kindly advise you to fix this issue since the previous reviewer didn't notice this.

Reply:
Hello, @FrederickMih  Thanks for your feedback, I appreciate your time. Please I would love it if you could throw more light on this part of the review. The search input is not a requirement in the project, It's just a code I added to make my work better. line 46 to 49 above is not where the data was filtered. It was filtered in line 22. Please throw more light so I can be clear.

Reply:
Thanks for the review generally, I will do well to write a test for the pure functions in redux

Reply:
Hi @Ibizugbe, it's a requirement of this project to Filter data retrieved from the API. Filtering should obviously be done on an input search field. Check out the third point on the technical requirements listing screenshot below:

![Screenshot from 2022-09-03 19-11-12](https://user-images.githubusercontent.com/63087491/188283239-a167d499-5335-4a30-9e5f-42211b7880b1.png)